### ISSUE TITLE
Failed to reproduce the performance of YOLOV11 on COCO.

### ISSUE THREAD

**Issue by sunsmarterjie** [ultralytics/ultralytics#17165]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Thank you for your excellent work. However, when I tried to reproduce the performance of YOLO11-S on COCO (training-from-scratch), it seems that the current code cannot reproduce the results you reported (only get 46.0, so 1.0 lower than you reported). I only did the following:

1. Downloaded the code and set up the environment.
2. Modified the configuration file according to previous issues (correct the augmentations).
3. Executed the code:
“
model = YOLO("yolo11s.yaml")
train_results = model.train(
    data="coco.yaml",  # path to dataset YAML
    epochs=600,  # number of training epochs
    imgsz=640,  # training image size
    device="0,1,2,3",  # device to run on, e.g., device=0 or device=0,1,2,3 or device=cpu
)
"

Here is the log file from my run, along with the corresponding configuration details (I copy the args.yaml -> args.txt, so that I can upload it). Could you explain the reason and provide some suggestions? Also, could you share the log file from your training?

[args.txt](https://github.com/user-attachments/files/17522947/args.txt)
[results.csv](https://github.com/user-attachments/files/17523058/results.csv)

**Comment by UltralyticsAssistant** [comment]
👋 Hello @sunsmarterjie, thank you for your interest in Ultralytics 🚀! We appreciate your effort in trying to reproduce the results of YOLO11-S on COCO. An Ultralytics engineer will assist you soon.

In the meantime, I'd like to provide some resources that might help. We recommend visiting the [Docs](https://docs.ultralytics.com) for a wealth of information on setting up and using YOLO models, including [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples.

If this is a 🐛 Bug Report and you haven't done so already, please ensure your test includes a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to assist with troubleshooting.

For your custom training ❓ question, detailed information including dataset samples and training logs can be very helpful. Ensure you follow our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/) to fine-tune your configuration.

Join the Ultralytics community on platforms that suit you best. Engage in real-time discussions on [Discord](https://ultralytics.com/discord) 🎧, share insights on [Discourse](https://community.ultralytics.com), or contribute to discussions on our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

To make sure you’re working with the latest improvements and fixes, please upgrade to the latest `ultralytics` package with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can be run seamlessly in several up-to-date environments, complete with all necessary dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), and [PyTorch](https://pytorch.org/):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VMs. Follow the [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMIs. Follow our [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. There’s a guide for Docker [here](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If the badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are currently passing. These tests ensure that YOLO performs correctly across various [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Stay tuned for further assistance from our team! 😊

**Comment by sunsmarterjie** [comment]
If you need other information, please let me know.

**Comment by glenn-jocher** [comment]
@sunsmarterjie official training curves and arguments for YOLO11s are at https://hub.ultralytics.com/models/Pi3uAeiO8JYD2bX0wotZ
![Uploading Screenshot 2024-10-25 at 16.38.33.png…]()

**Comment by sunsmarterjie** [comment]
Thanks for your reply, it seems like I forget modify the "scale". I will re-run it. Thanks!

**Comment by glenn-jocher** [comment]
You're welcome! If you have any more questions or need further assistance, feel free to ask. Good luck with your re-run!

**Comment by sunsmarterjie** [comment]
I finally get a result of 46.5, still 0.5 lower than you reported

**Comment by sunsmarterjie** [comment]
I am wondering how many times do you run and get the result? I mean is it possible that this falls within the margin of error?

**Comment by glenn-jocher** [comment]
It's possible that slight variations in results can occur due to randomness in training. Running multiple times and averaging results can help confirm consistency.

**Comment by Y-T-G** [comment]
@sunsmarterjie You have 4 GPUs and you're using batch size of 128 which makes it 128/4 = 32. The original model was trained with a single GPU with a batch of 128.

**Comment by sunsmarterjie** [comment]
How many GPUs do you use to train the model? What is the total batch size?

**Comment by sunsmarterjie** [comment]
So you guys trained the model using only 1 gpu?  Even that, do you think there is a difference?

**Comment by Y-T-G** [comment]
Your batch size is different, so results will be different

**Comment by sunsmarterjie** [comment]
Hi, guys, I try to reproduce the result of YOLOv11-N on COCO2017 using the completely same configurations, but only get a 39.0 mAP (0.5 lower than you reported). Here is the detailed log file and configurations. Looking forward to hearing from you to get me some advice...

[results.csv](https://github.com/user-attachments/files/17856115/results.csv)
[args.txt.txt](https://github.com/user-attachments/files/17856118/args.txt.txt)

**Comment by sunsmarterjie** [comment]
I am also running other scale models and I will let you know my reproduced results.

**Comment by glenn-jocher** [comment]
Thank you for your efforts in reproducing the results. Minor variations can occur due to factors like initialization and randomness in training. Comparing your configurations closely with the official ones and ensuring consistent settings, especially in data preprocessing and augmentation, can help. Let us know if you have further questions or updates!

**Comment by sunsmarterjie** [comment]
I have carefully compared the configurations (see my uploaded args.txt), and I do not modify anything after downloading the repo. So I do not think there are differences.

**Comment by sunsmarterjie** [comment]
Could you please release the training details? I think that will significantly help the community to reproduce the reported results?

**Comment by glenn-jocher** [comment]
Thank you for your suggestion. The official training details, including arguments and curves, are already available on the [Ultralytics HUB](https://hub.ultralytics.com/). Please refer to them to ensure consistent reproduction of results.

**Comment by glenn-jocher** [comment]
Thank you for pointing this out. The settings on the Ultralytics HUB are accurate and reflect the configurations used for training YOLO11n. If you find conflicting information elsewhere, please prioritize the HUB configurations. Let me know if you observe specific discrepancies, and I'll help clarify.

**Comment by sunsmarterjie** [comment]
Using the HUB configurations only leads to a 37.3mAP (log:
[results.csv](https://github.com/user-attachments/files/18031611/results.csv); config:
[args.txt](https://github.com/user-attachments/files/18031637/args.txt)). It seems that the configurations here (https://github.com/ultralytics/ultralytics/issues/16607#issuecomment-2388336537) is closer to the reproducible configuration.

I also train YOLO11-S/M models using the configurations here (https://github.com/ultralytics/ultralytics/issues/16607#issuecomment-2388336537) and all the results are slower than the reported ones. (YOLO11-S's log:
[results.csv](https://github.com/user-attachments/files/18031677/results.csv) and config:
[args.txt](https://github.com/user-attachments/files/18031679/args.txt). YOLO11-M's log:
[results.csv](https://github.com/user-attachments/files/18031697/results.csv)
 and config:
[args.txt](https://github.com/user-attachments/files/18031696/args.txt)
)

**Comment by ZZxiaoxiao** [comment]
funny

**Comment by glenn-jocher** [comment]
Thank you for your updates and for pointing out the discrepancies. The configurations on the Ultralytics HUB represent official details, and any inconsistencies are worth investigating. For further clarification regarding augmentations or settings, I recommend referring to updates in the [documentation](https://docs.ultralytics.com) or reaching out via GitHub Issues for coordinated follow-up. Let us know how else we can assist!

**Comment by sunsmarterjie** [comment]
If the configurations on the HUB are officially trained ones, it’s very strange that only YOLO11-S used copy_paste/mixup/mosaic, while YOLO11-N/M/L/X didn’t use any of them (all set to 0s).

**Comment by glenn-jocher** [comment]
Thank you for your observation. The discrepancy regarding augmentation settings has been noted, and I recommend following the official HUB configurations for consistency. If unexpected behavior persists, please document this in an issue so the team can further investigate.
